components/atom/popover/src/index.js

```diff
@@ -1,99 +1,120 @@
 import {
-  lazy,
   Children,
   cloneElement,
   useEffect,
   useState,
   useRef,
-  Suspense
+  forwardRef
 } from 'react'
 import PropTypes from 'prop-types'
-import {PLACEMENTS} from './config'
+import {useControlledState} from '@s-ui/react-hooks'
+import loadable from '@loadable/component'
+import {PLACEMENTS, TRIGGERS} from './config'
 
 const BASE_CLASS = 'sui-AtomPopover'
 const CLASS_INNER = `${BASE_CLASS}-inner`
 const PREFIX_PLACEMENT = `${BASE_CLASS}-`
 const DEFAULT_OFFSET = 'auto,4px'
-const DEFAULT_TRIGGER = 'legacy'
+const DEFAULT_TRIGGER = TRIGGERS.LEGACY
 const DEFAULT_DELAY = 0
 
-const Popover = lazy(() => import('reactstrap/lib/Popover'))
+// https://github.com/reactstrap/reactstrap/blob/7.1.0/src/TooltipPopoverWrapper.js
+const Popover = loadable(() => import('reactstrap/lib/Popover'), {ssr: true})
 
-function AtomPopover({
-  children,
-  closeIcon,
-  disableNativeToggle,
-  content,
-  id,
-  onClose = () => {},
-  onOpen = () => {},
-  placement = PLACEMENTS.BOTTOM,
-  showPopover,
-  hideArrow = true
-}) {
-  const targetRef = useRef()
-  const [internalShowPopover, setInternalShowPopover] = useState(showPopover)
+const adaptDeprecatedProps = ({showPopover, isVisible, ...props}) => {
+  return {
+    isVisible: isVisible === undefined ? showPopover : isVisible,
+    ...props
+  }
+}
 
+const AtomPopover = forwardRef((props, outRef) => {
+  const {
+    children,
+    closeIcon,
+    content,
+    onClose = () => {},
+    onOpen = () => {},
+    placement = PLACEMENTS.BOTTOM,
+    isVisible,
+    defaultIsVisible,
+    hideArrow = true,
+    trigger = DEFAULT_TRIGGER
+  } = adaptDeprecatedProps(props)
+  const targetRef = useRef()
+  const [ready, setReady] = useState()
+  const [isVisibleState, setIsVisibleState] = useControlledState(
+    isVisible,
+    defaultIsVisible
+  )
   useEffect(() => {
-    setInternalShowPopover(showPopover)
-  }, [showPopover])
+    setReady(targetRef.current)
+  }, [targetRef, setReady])
 
   const extendChildren = () => {
-    const onClick = () => {
+    const onClick = (onClickHandler = () => null) => ev => {
       onOpen()
-      setInternalShowPopover(true)
+      setIsVisibleState(true)
+      onClickHandler(ev)
     }
-    const ref = targetRef
-    const childrenOnly = Children.only(children)
-    return Children.map(childrenOnly, child => {
+    const childrenOnly =
+      typeof children === 'string'
+        ? [<span key={1}>{children}</span>]
```

Comment:
why do you need to wrap the string children in a span?

Reply:
if we do not, children values like direct string (text nodes) or other elements crashes

Reply:
I saw it now, it is in order to attach the attrs (with the onClick) to it 👍🏻